spring-cloud-skipper-shell-commands/src/main/java/org/springframework/cloud/skipper/shell/command/PackageCommands.java

```diff
@@ -49,10 +65,60 @@ public PackageCommands(SkipperClient skipperClient) {
 		this.skipperClient = skipperClient;
 	}
 
-	@ShellMethod(key = "package list", value = "Get the package metadata")
-	public String packageMetadata(
-			@ShellOption(help = "boolean to set for more detailed package metadata", defaultValue = "false") boolean details) {
-		return skipperClient.getPackageMetadata(details);
+	@ShellMethod(key = "package search", value = "Search for the packages")
+	public Object searchPackage(
+			@ShellOption(help = "wildcard expression to search for the package name", defaultValue = NULL) String name,
+			@ShellOption(help = "boolean to set for more detailed package metadata", defaultValue = "false") boolean details)
+			throws JsonProcessingException {
+		PagedResources<PackageMetadataResource> resources = skipperClient.getPackageMetadata(name, details);
+		if (!details) {
+			LinkedHashMap<String, Object> headers = new LinkedHashMap<>();
+			headers.put("name", "Name");
+			headers.put("version", "Version");
+			headers.put("description", "Description");
+			TableModel model = new BeanListTableModel<>(resources.getContent(), headers);
+			TableBuilder tableBuilder = new TableBuilder(model);
+			applyStyle(tableBuilder);
+			return tableBuilder.build();
+		}
+		else {
+			ObjectMapper mapper = new ObjectMapper();
+			String[][] data = new String[resources.getContent().size()][1];
+			TableModel model = new ArrayTableModel(data);
+			TableBuilder tableBuilder = new TableBuilder(model);
+			PackageMetadataResource[] packageMetadataResources = resources.getContent()
+					.toArray(new PackageMetadataResource[0]);
+			for (int i = 0; i < resources.getContent().size(); i++) {
+				for (int j = 0; j < 1; j++) {
+					data[i][j] = mapper.writeValueAsString(packageMetadataResources[i]);
+				}
+			}
+			return tableBuilder.build();
```

Comment:
You're not applying the style to the table, is that an omission?

Reply:
for the second condition, the style isn't applied as the content might be too long to handle. For now, we only apply the style if `details` isn't set.

Reply:
Don't you want at least borders?

Reply:
@ericbottard I tried to add the border there, but I stumble upon getting the alignment done correctly. I pushed some code changes. Could you help me figure out how this can be done correctly? Thanks!